### PR TITLE
Improve code quality across core logic classes with defensive coding and assertion practices (closes #128)

### LINKED ISSUE #128: Enhance Code Quality, Add Assertions, and Increase Test Coverage
This issue aims to:

Add meaningful assert statements in key classes (e.g., ParserUtil, UniquePersonList, AddressBook) to enforce assumptions.
Improve code quality by refactoring for SLAP, removing duplications, and ensuring defensive coding is consistently applied.
Identify under-tested methods or edge cases and add relevant test coverage, especially for parser and model logic.

This issue will guide the final polishing of code before v1.6.

Files targeted:
ParserUtil.java
ArgumentTokenizer.java
AddressBook.java
UniquePersonList.java
UniqueEventList.java

Test files: AddCommandParserTest.java, ArgumentTokenizerTest.java, ParserUtilTest.java

### PR DESCRIPTION
**Pull request by arshinsikka**
This PR enhances the code quality and robustness of key logic and model classes by applying defensive programming practices, Java assertions, and minor test cleanups, as part of the Week 11 quality pass.

🔧 Summary of Changes:
✅ Refactored ParserUtil.java to include defensive null checks and Java assertions in parsing methods without altering core logic behavior.
✅ Improved ArgumentTokenizer.java with defensive null guards, clarity in prefix extraction, and sorted logic to better handle unexpected inputs while preserving test behavior.
✅ Reviewed model classes AddressBook, UniquePersonList, and UniqueEventList:
Confirmed adherence to clean coding standards
Added clarifying comments and light improvements without altering logic

✅ Minor improvements in structure and readability for:
AddCommandParserTest.java
ArgumentTokenizerTest.java
ParserUtilTest.java
No major test logic was changed to avoid breaking valid existing coverage.

These improvements ensure maintainability and stability before the upcoming feature freeze.

✅ Closes #128

### CODE REVIEW

**Review by juneha1120 — APPROVED** [review]
Good work! LGTM!

File: src/main/java/trackup/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -175,23 +185,31 @@ public static String parseEventTitle(String eventTitle) throws ParseException {
...
      */
     public static Set<Index> parseContacts(Collection<String> indexes) throws ParseException {
         requireNonNull(indexes);
         final Set<Index> indexSet = new HashSet<>();
+
         for (String indexStr : indexes) {
-            indexSet.add(parseIndex(indexStr));
+            Index index = parseIndex(indexStr);
+            assert index.getZeroBased() >= 0 : "Parsed index should be non-negative";
+            indexSet.add(index);
```
Comment (caando) [inline src/main/java/trackup/logic/parser/ParserUtil.java:210]: Index object already cant be negative

File: src/main/java/trackup/model/AddressBook.java [hunk 2]
```diff
@@ -65,6 +67,8 @@ public void setEvents(List<Event> events) {
      */
     public void resetData(ReadOnlyAddressBook newData) {
         requireNonNull(newData);
+        assert newData != null : "New data for reset must not be null.";
```
Comment (caando) [inline src/main/java/trackup/model/AddressBook.java:70]: There is probably no need to add an assert != null immediately after requireNonNull

File: src/main/java/trackup/model/event/UniqueEventList.java [hunk 3]
```diff
@@ -55,6 +55,7 @@ public void add(Event toAdd) {
      */
     public void setEvent(Event target, Event editedEvent) {
         requireAllNonNull(target, editedEvent);
+        assert target != null && editedEvent != null;
```
Comment (caando) [inline src/main/java/trackup/model/event/UniqueEventList.java:58]: same here

Comment (arshinsikka) [reply src/main/java/trackup/model/event/UniqueEventList.java]: done!

File: src/main/java/trackup/logic/parser/ParserUtil.java (hunk 1 above)
Comment (arshinsikka) [reply src/main/java/trackup/logic/parser/ParserUtil.java]: Yes, removed the assert statement.

File: src/main/java/trackup/model/AddressBook.java (hunk 2 above)
Comment (arshinsikka) [reply src/main/java/trackup/model/AddressBook.java]: removed assert

**Review by caando — APPROVED** [review]
lgtm
